### ISSUE TITLE
[REQ] Allow ability to get raw response as String from ApiResponse

### ISSUE THREAD

**Issue by ericpetrowiakjelli** [OpenAPITools/openapi-generator#7019]
### Is your feature request related to a problem? Please describe.
I have run into clients that return void for various reasons, but actually have a text response. This is when using the generated client library 'jersey2'.
```
200
Content-disposition: attachment;filename=blah.csv 
Content-Length: 36
Content-Type: text/csv
blahblahblah
```

## Describe the solution you'd like
I'd like for the ApiResponse<T> to have an additional field that would return the raw response as a String.

## Describe alternatives you've considered
No alternative, short of updating annotations where api is defined to potentially get expected response.

## Additional context
Currently using version 4.3.0

**Comment by 0x7162** [comment]
@spacether 

I second the issue,
i.e. for my spec file:
```
    "/bla/endpoint" : {
      "get" : {
        "operationId" : "bla_bla",
        "responses" : {
          "default" : {
            "description" : "default response",
            "content" : {
              "text/csv" : { }
            }
          }
        }
      }
    },
```

I'm getting in my generated api function:
```
        return self.api_client.call_api(
            '/bla/endpoint', 'GET',
            ...
            response_type=None,  # noqa: E501
```

which causes data to become empty (`api_client.py`):
```
        # deserialize response data
        if response_type:  #  <---- HERE we jump to the next block
            if response_type != (file_type,):
                encoding = "utf-8"
                content_type = response_data.getheader('content-type')
                if content_type is not None:
                    match = re.search(r"charset=([a-zA-Z\-\d]+)[\s\;]?", content_type)
                    if match:
                        encoding = match.group(1)
                response_data.data = response_data.data.decode(encoding)

            return_data = self.deserialize(
                response_data,
                response_type,
                _check_type
            )
        else:
            return_data = None #      <---- HERE we get empty data
```

**Comment by spacether** [comment]
How about using python-experimental as your client?
Every time a response is returned, the raw response is accessible as [api_response.response](https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python-experimental/petstore_api/paths/store_order/post.py#L44)
Per:
```
@dataclass
class ApiResponseFor200(api_client.ApiResponse):
    response: urllib3.HTTPResponse
    body: typing.Union[
        SchemaFor200ResponseBodyApplicationXml,
        SchemaFor200ResponseBodyApplicationJson,
    ]
    headers: schemas.Unset = schemas.unset

class PlaceOrder(BaseApi):
    # this class is used by api classes that refer to endpoints with operationId fn names

    def place_order(
        self: BaseApi,
        body: typing.Union[SchemaForRequestBodyApplicationJson],
        content_type: str = 'application/json',
        accept_content_types: typing.Tuple[str] = _all_accept_content_types,
        stream: bool = False,
        timeout: typing.Optional[typing.Union[int, typing.Tuple]] = None,
        skip_deserialization: bool = False,
    ) -> typing.Union[
        ApiResponseFor200,
        api_client.ApiResponseWithoutDeserialization
    ]:
        return self._place_order(
            body=body,
            content_type=content_type,
            accept_content_types=accept_content_types,
            stream=stream,
            timeout=timeout,
            skip_deserialization=skip_deserialization
        )
```

**Comment by 0x7162** [comment]
> How about using python-experimental as your client? Getting the raw response from the result of the endpoint is a feature of that generator.
> 

Well, I have a bunch of issues for `python-experimental`

First, it requires python3.9 (so it doesn't scale for all the consumers)
Second, it may generate such a code:
```
        def M(cls):
            return cls("COUNT_1M")
        
        @classmethod
        @property
        def 0M(cls):
            return cls("COUNT_50M")
```
which causes naming issues:
```
    def 0M(cls):
        ^
SyntaxError: invalid syntax
```


it is structure like that in `python`:
```
    allowed_values = {
...
        ('cell_count',): {
            '1M': "COUNT_1M",
            '50M': "COUNT_50M",
```

so I consider `python-experimental` unstable.

Any other solutions?

So far I have the solution to modify a spec like that:
```
                content:
                    text/csv:
                        schema:
                            type: string
```

**Comment by spacether** [comment]
Sorry to hear that there are a number of issues that are blocking your usage of python experimental.
Could you open another ticket specifying them?
What is the highest version of python that is viable for you to use?
I want to make sure that it meets our users needs.
Looking at your enum, it looks like the issue that you mentioned (invalid enum names) exists in an older version of the generator. When I use latest master I see the below working code:
```
class EnumClass(
    schemas.SchemaEnumMakerClsFactory(
        enum_value_to_name={
            "_abc": "_ABC",
            "-efg": "EFG",
            "(xyz)": "XYZ",
            "COUNT_1M": "COUNT_1M",
            "COUNT_50M": "COUNT_50M",
        }
    ),
    schemas.StrSchema
):
...
    @classmethod
    @property
    def COUNT_1M(cls):
        return cls("COUNT_1M")

    @classmethod
    @property
    def COUNT_50M(cls):
        return cls("COUNT_50M")
```

In terms of solutions that will work with the `python` generator, how about passing in `_preload_content` when calling the endpoint? Then the raw response will be returned.

**Comment by 0x7162** [comment]
I'm using the latest (6.0.1) from
https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/6.0.1/
I assume it doesn't make sense to create a ticket - I assume it is already fixed and it will be no factor in 6.0.2+ release.

`_preload_content=False` worked out, thanks!
I assume we can close the ticket?
@ericpetrowiakjelli  please consider `_preload_content` solution provided by @spacether

**Comment by spacether** [comment]
The python client in v6.2.0 and onward requires python 3.7 and higher and the raw response is accessible as a property on every endpoint api response
